main.go

```diff
@@ -214,11 +264,11 @@ func validateEvent(event *types.Event) error {
 	}
 
 	if err := event.Entity.Validate(); err != nil {
-		return err
+		log.Fatal(err.Error())
```

Comment:
Calling log.Fatal causes the program to log and then immediately exit. This makes the function hard to test, because the whole runtime process will get taken down, including the test runner. It also prevents resources from being cleaned up by defers.

Generally, only call log.Fatal in main().